sidechain/src/lib.rs

```diff
@@ -0,0 +1,221 @@
+/*
+Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use codec::Encode;
+use frame_support::{
+	dispatch::DispatchResultWithPostInfo,
+	traits::{Currency, Get},
+};
+use frame_system::{self};
+use pallet_teerex::Pallet as Teerex;
+use sidechain_primitives::*;
+use sp_core::H256;
+use sp_std::{prelude::*, str};
+use teerex_primitives::ShardIdentifier;
+
+pub use crate::weights::WeightInfo;
+
+// Disambiguate associated types
+pub type AccountId<T> = <T as frame_system::Config>::AccountId;
+pub type ShardBlockNumber = (ShardIdentifier, u64);
+
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	#[pallet::without_storage_info]
+	pub struct Pallet<T>(PhantomData<T>);
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config + pallet_teerex::Config {
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+		type Currency: Currency<<Self as frame_system::Config>::AccountId>;
+		type MomentsPerDay: Get<Self::Moment>;
+		type WeightInfo: WeightInfo;
+		type MaxSilenceTime: Get<Self::Moment>;
```

Comment:
We don't need these types here.

Reply:
You mean Currency, MomentPerDay, MaxDilenceTime or also WeightInfo?

Reply:
Yes.